src/main/java/org/terasology/climateConditions/alterationEffect/BodyTemperatureAlterationSystem.java

```diff
@@ -0,0 +1,120 @@
+import org.terasology.logic.delay.DelayedActionTriggeredEvent;
+import org.terasology.registry.In;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * This system manages the effects of the Body Temperature Alteration Effect.
+ */
+@RegisterSystem
+public class BodyTemperatureAlterationSystem extends BaseComponentSystem {
+
+    @In
+    Context context;
+
+    /**
+     * This will store the mapping of the effect constants to the effect components.
+     */
+    private Map<String, Class<? extends Component>> effectComponents = new HashMap<>();
+
+    /**
+     * This will store the mapping of the effect constants to the alteration effects.
+     */
+    private Map<String, AlterationEffect> alterationEffects = new HashMap<>();
+
+    @Override
+    public void initialise() {
+        effectComponents.put(AlterationEffects.BODY_TEMPERATURE, AffectBodyTemperatureComponent.class);
+        alterationEffects.put(AlterationEffects.BODY_TEMPERATURE, new BodyTemperatureAlterationEffect(context));
+    }
+
+    /**
+     * When the body temperature of an entity changes, modify the change in temperature based on what effects are being
+     * applied to the entity.
+     *
+     * @param event Stores information on the body temperature change that is to be applied and collects
+     *         potential modifiers.
+     * @param entityRef The entity experiencing a change in body temperature.
+     */
+    @ReceiveEvent
+    public void modifyTemperatureChange(AffectBodyTemperatureEvent event, EntityRef entityRef) {
+        // If the entity's stunned, prevent it from moving.
+        AffectBodyTemperatureComponent affectBodyTemperatureComponent =
+                entityRef.getComponent(AffectBodyTemperatureComponent.class);
+        if (affectBodyTemperatureComponent != null) {
+            if (affectBodyTemperatureComponent.id.equals("default")) {
+                event.multiply(affectBodyTemperatureComponent.multiplier);
+            } else {
+                if (affectBodyTemperatureComponent.id.equals("activateWhenEnvironmentColder") && event.isNegative()) {
+                    event.multiply(affectBodyTemperatureComponent.multiplier);
+                } else if (affectBodyTemperatureComponent.id.equals("activateWhenEnvironmentHotter") && !event.isNegative()) {
+                    event.multiply(affectBodyTemperatureComponent.multiplier);
+                }
+            }
+        }
+    }
+
+    /**
+     * Once a basic effect's duration has expired, remove the effect from the entity that had it, send a removal event,
+     * informing the other effect systems, and then re-apply the associated alteration effect.
+     *
+     * @param event Event with information of what particular effect expired.
+     * @param entity The entity that had the expired effect.
+     */
+    @ReceiveEvent
+    public void expireEffects(DelayedActionTriggeredEvent event, EntityRef entity) {
```

Comment:
Actually, there is an exact copy of this method in alteration effects, this was essential in order to avoid AE depending on CC. But I'm not sure, should I make some changes so that it reacts only to the effects within CC?